tensorflow/java/src/gen/cc/source_writer.cc

```diff
@@ -83,20 +83,20 @@ SourceWriter& SourceWriter::Append(const StringPiece& str) {
 }
 
 SourceWriter& SourceWriter::AppendType(const Type& type) {
-  if (type.kind() == Type::Kind::GENERIC && type.name().empty()) {
+  if (type.unknown()) {
     Append("?");
   } else {
     Append(type.name());
-  }
-  if (!type.parameters().empty()) {
-    Append("<");
-    for (const Type& t : type.parameters()) {
-      if (&t != &type.parameters().front()) {
-        Append(", ");
+    if (!type.parameters().empty()) {
+      Append("<");
+      for (const Type& t : type.parameters()) {
+        if (&t != &type.parameters().front()) {
```

Comment:
The intention here is to avoid the first one, right? How about:

```c++
bool first = true;
for (const Type& t: type.parameters()) {
  if (!first) Append(", ");
  AppendType(t);
  first = false;
}
```